packages/dialect-react-ui/components/Chat/screens/Main/pages/ThreadPage/MessageInput.tsx

```diff
@@ -41,14 +44,21 @@ export default function MessageInput({
                 onKeyDown={onEnterPress}
                 placeholder="Write something"
                 className={clsx(textArea, 'dt-resize-none dt-h-full dt-w-full')}
+                disabled={disableTextarea}
               />
               <ButtonBase
                 className="dt-button dt-absolute dt-inset-y-0 dt--right-2 dt-flex dt-items-center dt-pr-3 disabled:dt-cursor-not-allowed"
-                disabled={disabled}
+                disabled={disableSendButton}
               >
-                <icons.arrowsmright
-                  className={clsx(sendButton, disabled ? 'dt-opacity-50' : '')}
-                />
+                {
+                  disableTextarea ? (
```

Comment:
this name is a bit misleading here but thinking on it I'm not sure what's a better set of `bool`s. most explicit set would be

- `buttonDisabled`
- `inputDisabled`
- `sending`
- `creating`

maybe overkill though

Reply:
I'll go with `inputDisabled`